src/component/assignment-4/SubmitValidation.jsx

```diff
@@ -0,0 +1,120 @@
+"use client";
+import { Box, Button, TextField, Alert } from "@mui/material";
+import { useState } from "react";
+
+const SubmitValidation = () => {
+  const [userName, setUserName] = useState("");
+  const [password, setPassword] = useState("");
+  const [confPassword, setConfPassword] = useState("");
+  const [isPasswordWrong, setIsPasswordWrong] = useState(false);
+  const [isUserNameWrong, setIsUserNameWrong] = useState(false);
+  const [passwordMatch, setPasswordMatch] = useState(true);
+  const [errorSummary, setErrorSummary] = useState([]);
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+
+   
+    setIsUserNameWrong(false);
+    setIsPasswordWrong(false);
+    setPasswordMatch(true);
+    setErrorSummary([]);
+
+    const errors = [];
+
+   
+    if (userName.trim().length < 2) {
+      setIsUserNameWrong(true);
+      errors.push("Username must be at least 2 characters long.");
+    }
+
+   
+    if (
```

Comment:
Unnecessary state are created, Use single state with object to handle error validation.